lib/active_resource/base.rb

```diff
@@ -436,6 +471,16 @@ def schema=(the_schema)
           # purposefully nulling out the schema
           @schema = nil
           @known_attributes = []
+
+          # TODO uses private APIs -- mostly for test harness resetting
+          undefine_attribute_methods
+          if respond_to?(:reset_default_attributes)
+            reset_default_attributes
+          elsif respond_to?(:attribute_types=) && respond_to?(:_default_attributes=)
+            self.attribute_types = Hash.new(ActiveModel::Type.default_value)
+            self._default_attributes = ActiveModel::AttributeSet.new({})
+          end
```

Comment:
This block is an attempt to reverse any `attribute`-defined methods. 

The current attempt is messy and relies on private APIs. The conditionals aim to support the range of Active Model versions.

I hope there's a better way to reverse these declarations -- ideally in a test harness-only way, since the main motivation is to support the `setup` and `teardown` blocks in https://github.com/rails/activeresource/blob/main/test/cases/base/schema_test.rb.

Does Active Resource explicitly support calling `.schema { ... }` or `.schema=` multiple times for the same `ActiveResource::Base` subclass?

Reply:
CI tests are flaky across the matrix because teardown isn't entirely working and state is leaking between runs.